trollmoves/tests/test_client.py

```diff
@@ -701,98 +751,223 @@ def test_reload_config(Listener, NoisyPublisher):
+def test_reload_config_providers_removed(Listener, NoisyPublisher, client_config_1_item,
+                                         client_config_1_item_two_providers):
+    """Test trollmoves.client.reload_config() when providers are removed."""
+    from trollmoves.client import reload_config
+
+    listener = MagicMock()
+    Listener.return_value = listener
+
+    chains = {}
+    callback = MagicMock()
+
+    try:
+        reload_config(client_config_1_item, chains, callback=callback)
+        num_providers = len(chains["eumetcast_hrit_0deg_scp_hot_spare"]._config['providers'])
+        assert len(chains["eumetcast_hrit_0deg_scp_hot_spare"].listeners) == num_providers
+        assert Listener.call_count == num_providers
+        reload_config(client_config_1_item_two_providers, chains, callback=callback)
+        num_providers2 = len(chains["eumetcast_hrit_0deg_scp_hot_spare"]._config['providers'])
+        assert num_providers2 != num_providers
+        assert len(chains["eumetcast_hrit_0deg_scp_hot_spare"].listeners) == num_providers2
+        assert Listener.call_count == num_providers
+        assert listener.stop.call_count == num_providers - num_providers2
+    finally:
+        for key in chains:
+            try:
+                chains[key].stop()
+            except AttributeError:
+                pass
+        os.remove(client_config_1_item)
+        os.remove(client_config_1_item_two_providers)
+
+
+@patch('trollmoves.client.NoisyPublisher')
+@patch('trollmoves.client.Listener')
+def test_reload_config_provider_topic_changed(Listener, NoisyPublisher, client_config_1_item,
+                                              client_config_1_item_topic_changed):
+    """Test trollmoves.client.reload_config() when the message topic is changed."""
+    from trollmoves.client import reload_config
+
+    chains = {}
+    callback = MagicMock()
+
+    try:
+        reload_config(client_config_1_item, chains, callback=callback)
+        num_providers = len(chains["eumetcast_hrit_0deg_scp_hot_spare"]._config['providers'])
+        assert Listener.call_count == num_providers
+        reload_config(client_config_1_item_topic_changed, chains, callback=callback)
+        assert Listener.call_count == 2 * num_providers
+        num_providers2 = len(chains["eumetcast_hrit_0deg_scp_hot_spare"]._config['providers'])
+        assert num_providers2 == num_providers
+        assert len(chains["eumetcast_hrit_0deg_scp_hot_spare"].listeners) == num_providers2
+    finally:
+        for key in chains:
+            try:
+                chains[key].stop()
+            except AttributeError:
+                pass
+        os.remove(client_config_1_item)
+        os.remove(client_config_1_item_topic_changed)
+
+
+@patch('trollmoves.client.Chain')
+def test_reload_config_chain_not_recreated(Chain, client_config_1_item, client_config_1_pub_item_modified):
+    """Test that Chain is not recreated when existing chain is modified."""
```

Comment:
I'm not sure I understand this sentence...

Reply:
Clarified. I hope.